lib/features/exhibition_flow/screens/voucher_code_screen.dart

```diff
@@ -0,0 +1,129 @@
+    return BlocConsumer<AuthCubit, AuthState>(
+      listener: (context, authState) {
+        if (authState is ExternalErrorState) {
+          _showErrorMessage();
+        } else if (authState is LoggedInState) {
+          context
+              .read<ProfilesCubit>()
+              .fetchProfiles(authState.session.userGUID);
+        }
+      },
+      builder: (context, authState) {
+        return BlocConsumer<ProfilesCubit, ProfilesState>(
+          listener: (context, propfilesState) {
+            if (propfilesState is ProfilesExternalErrorState) {
+              _showErrorMessage();
+            } else if (propfilesState is ProfilesUpdatedState &&
+                propfilesState.activeProfile == Profile.empty()) {
+              context
+                  .read<ProfilesCubit>()
+                  .setActiveProfile(propfilesState.profiles[0]);
+              context.read<FlowsCubit>().startExhibitionFlow();
+
+              context.pushNamed(Pages.scanNFC.name);
+            }
+          },
+          builder: (context, propfilesState) {
+            return Scaffold(
+              body: authState is LoadingState ||
+                      propfilesState is ProfilesLoadingState
+                  ? const Center(child: CircularProgressIndicator())
+                  : SafeArea(
+                      child: SizedBox.expand(
+                        child: SingleChildScrollView(
+                          child: Column(
+                            mainAxisSize: MainAxisSize.min,
+                            crossAxisAlignment: CrossAxisAlignment.center,
+                            children: [
+                              SizedBox(height: size.height * 0.1),
+                              Text(
+                                "Let's begin!",
+                                style: Theme.of(context)
+                                    .textTheme
+                                    .titleLarge
+                                    ?.copyWith(fontWeight: FontWeight.w700),
+                              ),
+                              const SizedBox(height: 5),
+                              Text(
+                                'Enter your unique Givt code',
+                                style: GoogleFonts.mulish(
+                                  textStyle:
+                                      Theme.of(context).textTheme.titleLarge,
+                                  fontWeight: FontWeight.w500,
+                                ),
+                              ),
+                              SizedBox(height: size.height * 0.15),
+                              VoucherCodeInput(
+                                onChanged: _updateVoucherCode,
+                              ),
+                              SizedBox(height: size.height * 0.10),
+                            ],
+                          ),
+                        ),
+                      ),
+                    ),
+              floatingActionButtonLocation:
```

Comment:
Also for this one. Most places we shouldn't use a floating button as it will go over content sometimes unintentionally (for examples with smaller screens, when de keyboard popups - did you test that?).

Reply:
there is no much content to scroll, and I've covered it with ScrollView, so should work fine I think :). Also the exhibition flow should work on our (Givt) devices as far as I understood. However let's test since did not test it an all many screens of course